### PR TITLE
Additional checks for php 8 array keys.

### PR DESCRIPTION
**Pull request by guelzow**
Hi everyone.

I've added a few additional checks for php 8 array keys.
These are the ones I was able to identify so far. I'm pretty sure that I missed some of them.

Please test, review and comment.   😃 

Tobi

### CODE REVIEW

**Comment by aimeos** [comment]
You can always use the `??` operator to test if the value is set. This will simplify your code a lot

**Comment by guelzow** [comment]
I'm sorry for the late reply, but a few work related things came up.

I'll see what I can do over the weekend.

**Comment by guelzow** [comment]
Huh, I was not aware t hat I could stack null coalescing operators.

So far I was trying to stay away from ternary operators, because I've seen people nesting them to make their code illegible on purpose.

File: Resources/Private/Config/resource.php [hunk 1]
```diff
@@ -2,19 +2,21 @@
 
 $publicPath = \TYPO3\CMS\Core\Core\Environment::getPublicPath();
 
+$defaultConnection = $GLOBALS['TYPO3_CONF_VARS']['DB']['Connections']['Default'];
+$fallbackConnection = $GLOBALS['TYPO3_CONF_VARS']['DB'];
```
Comment (aimeos) [inline Resources/Private/Config/resource.php:6]: As Aimeos only supports TYPO3 10 and 11, you can remove the fallback now. Then, you don't need to stack the `??` operator

**Comment by aimeos** [comment]
Thank you very much!

**Comment by guelzow** [comment]
As always pleasure.

**Pull request merged**